### PR TITLE
GameDB: Add game fixes for Silent Hill 2 and 3

### PR DESCRIPTION
**Pull request by FlatOutPS2**
Adds FMVinSoftwareHack to the game fixes list and activates it for Silent Hill 2 and 3.

Closes https://github.com/PCSX2/pcsx2/issues/1787

### CODE REVIEW

**Comment by ssakash** [comment]
[**Two years ago**](https://github.com/PCSX2/pcsx2/pull/488#issuecomment-89002511), @ramapcsx2  was against adding this hack to the GameDB, not sure if he's fine with it now.

**Comment by FlatOutPS2** [comment]
> Two years ago, @ramapcsx2 was against adding this hack to the GameDB, not sure if he's fine with it now.

The hack isn't perfect, but it works perfectly fine in affected games. Like most other hacks/game fixes, as long as it isn't added for all games it won't be an issue.

File: bin/GameIndex.dbf [hunk 1]
```diff
@@ -68,6 +68,7 @@
 -- DMABusyHack 		= 1 // Mana Khemia, Metal Saga. Denies writes to the DMAC when it's busy.
 -- VIFFIFOHack 		= 1 // Transformers Armada, Test Drive Unlimited. Fixes slow booting issue.
 -- VIF1StallHack 	= 1 // SOCOM II HUD and Spy Hunter loading hang.
+-- FMVinSoftwareHack= 1 // Silent Hill 2-3. Fixes FMVs that are obscured when using hardware rendering by switching to Software rendering while an FMV plays.
```
Comment (ssakash) [inline bin/GameIndex.dbf]: vertical alignment would be nice here.

Also ``Software`` **->** ``software``.

Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: It was perfectly aligned in notepad, but not in all text editors it seems. :p

Comment (FlatOutPS2) [inline bin/GameIndex.dbf]: It was caused by mixed up spaces and tabs, so I replaced all the tabs with spaces.

**Comment by gregory38** [comment]
would be nice to have a short explanation in the commit of the game issue. I guess 
* sh2 is upscaling line ?
* sh3 is gs memory wrapping ?

**Comment by FlatOutPS2** [comment]
> sh2 is upscaling line ?
sh3 is gs memory wrapping ?

Actually SH3 has the upscaling line issue as well as the memory wrapping issue.

**Comment by gregory38** [comment]
thanks you

**Comment by ramapcsx2** [comment]
I still dislike adding a convenience hack to automatic gamefixes. The hack produces totally unexpected behavior if it's automatically enabled. Instead, the user could be notified that this hack exists, should he want to have it.

**Comment by ramapcsx2** [comment]
In particular, this hack destroys the entire renderer context and creates a new one. This has implications to things like the display driver, 3rd party tools (FPS Meter, recording apps) and such. If it was "cleaner" somehow, it would be okay.

**Pull request merged**